pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -77,6 +86,77 @@ func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
 
 var _ webhook.CustomDefaulter = &PodWebhook{}
 
+func omitKueueLabels(l map[string]string) map[string]string {
+	result := map[string]string{}
+
+	for key, value := range l {
+		if !strings.HasPrefix(key, "kueue.x-k8s.io/") {
+			result[key] = value
+		}
+	}
+	return result
+}
+
+func containersShape(containers []corev1.Container) (result []map[string]interface{}) {
+	for _, c := range containers {
+		result = append(result, map[string]interface{}{
+			"image": c.Image,
+			"resources": map[string]interface{}{
+				"requests": c.Resources.Requests,
+			},
+			"ports": c.Ports,
+		})
+	}
+
+	return result
+}
+
+func getRoleHash(p *Pod) (string, error) {
+
+	shape := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"labels": omitKueueLabels(p.ObjectMeta.Labels),
+		},
+		"spec": map[string]interface{}{
+			"initContainers":            containersShape(p.Spec.InitContainers),
+			"containers":                containersShape(p.Spec.Containers),
+			"nodeSelector":              p.Spec.NodeSelector,
+			"affinity":                  p.Spec.Affinity,
+			"tolerations":               p.Spec.Tolerations,
+			"runtimeClassName":          p.Spec.RuntimeClassName,
+			"priority":                  p.Spec.Priority,
+			"preemptionPolicy":          p.Spec.PreemptionPolicy,
+			"topologySpreadConstraints": p.Spec.TopologySpreadConstraints,
+			"overhead":                  p.Spec.Overhead,
+			"volumes":                   p.Spec.Volumes,
+			"resourceClaims":            p.Spec.ResourceClaims,
+		},
+	}
+
+	shapeJson, err := json.Marshal(shape)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%x", sha256.Sum256(shapeJson)), nil
```

Comment:
I think I saw in other places that you were only taking 8 characters? Let's try to do it in one place, and this function seems like the best place.